src/CMakeLists.txt

```diff
@@ -1,5 +1,22 @@
+
+find_package(SISL)
```

Comment:
I would rather that this is controlled by an explicit option (i.e. WITH_SISL), to avoid having people getting an unexpectedly crippled base/types.

Reply:
> Perhaps we could always install `Spline.hpp` and `Trajectory.hpp`, but have it generate an `#error`, if SISL was not found?

This shifts the error to the compile phase instead of the configuration phase where it should occur.

In general I am fine with an option but it conflicts with having `optional=1` in the manifest.

Reply:
> In general I am fine with an option but it conflicts with having optional=1 in the manifest.

Not necessarily. Autoproj will have to set the flag if sisl is not excluded, which would be done with 

~~~
unless Autoproj.manifest.excluded?('external/sisl')

end
~~~

Within the package definition block [in rock.core's 02stable.autobuild](https://github.com/rock-core/package_set/blob/master/04stable.autobuild#L44)

Note that the `package_enabled?` call is actually deprecated since Autoproj v2. To facilitate the transition between v1 and v2, I disabled the deprecation warnings by default and then forgot I did :(